url/url.h

```diff
@@ -28,15 +28,61 @@ struct Host {
     std::variant<std::string, std::uint32_t, std::array<std::uint16_t, 8>> data;
 
     std::string serialize() const;
+
+    bool operator==(Host const &b) const {
```

Comment:
Looks like this can be defaulted?

Reply:
I thought so initially too, but it fails to compile.

```
./url/url.h: In member function 'bool url::Origin::operator==(const url::Origin&) const':
./url/url.h:60:44: error: no match for 'operator==' (operand types are 'const url::Host' and 'const url::Host')
   60 |             if (scheme == b.scheme && host == b.host && port == b.port) {
      |                                       ~~~~ ^~ ~~~~~~
      |                                       |         |
      |                                       |         const url::Host
      |                                       const url::Host
```

Reply:
nevermind, I didn't know this was a thing: https://en.cppreference.com/w/cpp/language/default_comparisons